snorkel/candidates.py

```diff
@@ -271,3 +283,30 @@ def apply(self, context, clear, split, check_for_existing=True, **kwargs):
 
             # Add Candidate to session
             yield self.candidate_class(**candidate_args)
+
+
+class OmniNgrams(Ngrams):
```

Comment:
So it looks like all this subclass is doing is checking for a specific `Context` type then iterating over its children... can we write this in a more general way?  I think having a specific class named `OmniNgrams` (which doesn't clearly delineate the relation to Ngrams / function)